app/controllers/orders_controller.rb

```diff
@@ -0,0 +1,32 @@
+class OrdersController < ApplicationController
+  def new
+    @order = current_client.orders.build
+    @order.delivery_addresses.build
+    @order.product = Product.find(params[:product_id])
+  end
+
+  def create
+    @order = current_client.orders.build(order_params)
+    @order.product = Product.find(params[:order][:product_id])
```

Comment:
This shouldn't be necessary, it's already on the `order_params`

Reply:
nice catch!